ospfd/ospf_abr.c

```diff
@@ -66,8 +66,11 @@ static void ospf_area_range_add(struct ospf_area *area,
 	apply_mask_ipv4(&p);
 
 	rn = route_node_get(ranges, (struct prefix *)&p);
-	if (rn->info)
+	if (rn->info){
 		route_unlock_node(rn);
+		ospf_area_range_free(range);
```

Comment:
This looks backwards to me.  Free the (rn->info) pointer w/ ospf_area_range_free(rn->info) and then set rn->info = range